core/src/services/rowService.js

```diff
@@ -0,0 +1,24 @@
+module.exports = (db) => {
+  return {
+    get(data) {
```

Comment:
I don't know much about `SQLite` prepare statements, but I _think_ the part that avoids SQL Injections is where the parameters are provided via the `?` operators.  In the next block of code we only pass `limit` and `page` via this mechanism.  I _think_ we need to pass all of the text blocks in this way.

Yes, we need to run the `prepare` without any `${}` fields, and only pass them in via the `.all` command.  Since most blocks are optional, I think we'll actually need separate sets of strings.  So for the `where` content we'll need one string (`WHERE`) that is provided to `prepare`, then another string where we provide `counter = "Germany"`.

Reply:
I think the way `prepare` keeps things secure is that it separates user generated (and possibly dodgy) content from system generated content. The system-generated content is passed to `prepare` to be built into SQL elements, with placeholders `?` for user generated content that is dropped into the relevant slot in the SQL string.

Reply:
@AbegaM and I have discussed this offline and are now in agreement on the strategy.